dashboard/app/public_json_api.go

```diff
@@ -45,20 +46,31 @@ func getExtAPIDescrForBugPage(bugPage *uiBugPage) *publicAPIBugDescription {
 			}
 			return []string{bugPage.Bug.ExternalLink}
 		}(),
-		FixCommits: func() []fixCommit {
+		FixCommits: func() []gitCommit {
 			if len(bugPage.Bug.Commits) == 0 {
 				return nil
 			}
-			var res []fixCommit
+			var res []gitCommit
 			for _, commit := range bugPage.Bug.Commits {
-				res = append(res, fixCommit{
+				res = append(res, gitCommit{
 					Title: commit.Title,
 					Hash:  commit.Hash,
 					Link:  commit.Link,
 				})
 			}
 			return res
 		}(),
+		CauseCommit: func() *gitCommit {
```

Comment:
FixCommits are generally provided manually by humans and high quality, while bisections are best effort and wrong quite often. More often then not bisections are inconclusive and have more then one possible Commit, for those this would always be empty.

Since this is and API, why not return the raw bisection info and let the consumer decide how to display it?

Reply:
It is a cause-commit, not a fix-commit.
I wanted to export "guilty" commit data and provide it only in the situations when we're sure who is guilty (to decrease false positive signals).

Reply:
Please reopen if you disagree.

Reply:
Yes that's exactly my point. I wouldn't call anything that comes out of a bisection "sure". On the other hands FixCommits (usually) don't come out of bisections, but are (usually) provided by a human.

Reply:
@HerrSpace what additional info do you propose to export?

I wouldn't export just all raw info we have and in exact object hierarchy we have. Exporting too much data and implementation details will tie us too much.

We have some flags related to how much we are sure about the commit, but we need to figure out how to export them for API in a reasonable way.

Reply:
How would this tie us down? An API gives us the freedom to share the information we currently can't or don't want to squeeze into the already loaded dashboard UI. I'd argue eventually the API should export all the information required to build the dashboard and more. Instead this merge request gives a more shallow view than what people can currently get from the dashboard, meaning they will likely have to scrape the HTML, which is something Florent has already done in the past.

For bisections I think it could just be a list with a least:
- type (cause, fix)
- list of commits (one for conclusive, multiple for inconclusive, none for everything else)
- bisect_log link
- error_log link (if present)
- isInvalid (InvalidatedBy is not nil)

This would make it easy for anybody to write quick scripts to analyse common patterns/issues in bisections. Even for us that's much nicer than having to interact with datastore directly.

Reply:
[citation needed] :) Modern app development is in large parts frontend code, consuming internal APIs with no stability guarantees. IMO the reason syzbot doesn't have something like that yet, is because we are building a site for a very conservative community that wouldn't much appreciate us using Angular / client site rendering.

We have seen a bunch of papers working with syzbot bugs. The only way the researchers can have collected these is by scraping HTML. I'm sure those people would have preferred using an API without guarantees over HTML. As long as we don't explicitly promise stability, it's well in our prerogative to change the API whenever we wish. This is not the kernel ABI.

Besides what I outlined is not exporting the raw jobs object with all fields anyhow, but a set of well established bisection details, no?

Reply:
for the CVE/OSV bit, only "confirmed" ones would be useful. I don't mind iterating over a list to find the confirmed ones, though.

Reply:
> Besides what I outlined is not exporting the raw jobs object with all fields anyhow, but a set of well established bisection details, no?

I was replying to "return the raw bisection info".

I have no concerns for an internal API for our own UI.